### ISSUE TITLE
Auto-Fill fails on SonicWall Firewalls

### ISSUE THREAD

**Issue by zimmbi** [keepassxreboot/keepassxc-browser#2169]
On SonicWall Gen7 Firewalls the "LOG IN" button isn't working after the chrome extensions fills the username and password fields.

## Possible Solution

Currently it works with this method, maybe it can be integrated in the extension?

`
 kpxc.setValueWithChange = function(field, value, forced = false) {
    if (!forced && field.readOnly) {
        return;
    }

    const simulateKeyEvent = function(elem, eventType, keyCode) {
        const event = new KeyboardEvent(eventType, {
            bubbles: true,
            cancelable: true,
            keyCode: keyCode,
            which: keyCode
        });
        elem.dispatchEvent(event);
    };

    const simulateInput = async function(elem, newValue) {
        elem.focus();
        for (let i = 0; i < newValue.length; i++) {
            const char = newValue.charAt(i);
            simulateKeyEvent(elem, 'keydown', char.charCodeAt(0));
            simulateKeyEvent(elem, 'keypress', char.charCodeAt(0));
            elem.value += char; // Add the character to the input value
            elem.dispatchEvent(new Event('input', { bubbles: true }));
            simulateKeyEvent(elem, 'keyup', char.charCodeAt(0));
            await new Promise(resolve => setTimeout(resolve, Math.random() * 40 + 10)); // Mimic human typing delay
        }
        elem.dispatchEvent(new Event('change', { bubbles: true }));
    };

    simulateInput(field, value);
};
`

## Steps to Reproduce (for bugs)
1. Go to SonicWall Firewall Login page (Gen 7)
2. Automatically fill username and password
3. Click "LOG IN"

**Comment by varjolintu** [comment]
I'm not really in favor to adding any delays that mimics typing etc. but I can take a look if this can be fixed in some other way.

**Comment by zimmbi** [comment]
any updates on this?

**Comment by varjolintu** [comment]
No. I would need to have an access to an actual login page for debugging this.

**Comment by zimmbi** [comment]
Ok but this are local firewalls, i can give you access to a test
environment if needed.

**Comment by varjolintu** [comment]
If you can do that, please provide any details in private via email.

**Comment by zimmbi** [comment]
> If you can do that, please provide any details in private via email.

mail sent

**Comment by ToniKoehler** [comment]
Vote on this feature! 👍
